File: app/src/main/java/huitca1212/cuantotemide/MainActivity.java
package huitca1212.cuantotemide;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.Toast;

public class MainActivity extends OptionsActivity implements View.OnClickListener {
	private Spinner countrySpinner;
	private Button startButton;
	private String[] dataArray = {"Selecciona país...",
			"Andorra", "Argentina", "Belice", "Bolivia", "Brasil", "Canadá", "Chile", "Colombia", "Costa Rica", "Cuba", "Ecuador", "El Salvador",
			"España", "Estados Unidos", "Francia", "Guatemala", "Guinea Ecuatorial", "Haití", "Honduras", "Marruecos", "México", "Nicaragua", "OTRO", "Panamá",
			"Paraguay", "Perú", "Puerto Rico", "República Dominicana", "Trinidad y Tobago", "Uruguay", "Venezuela"};
	private String countrySelected = dataArray[0];

	public static void startActivity(Activity activity) {
		Intent intent = new Intent(activity, MainActivity.class);
		activity.startActivity(intent);
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		countrySpinner = (Spinner)findViewById(R.id.country_spinner);
		startButton = (Button)findViewById(R.id.start_button);

		startButton.setOnClickListener(this);
		Utils.setAnalytics(this);
		setCountrySpinner();
	}

	private void setCountrySpinner() {
		ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, dataArray);
		countrySpinner.setAdapter(adapter);
		countrySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
			@Override
			public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
				countrySelected = dataArray[i];
			}

			@Override
			public void onNothingSelected(AdapterView<?> arg0) {
			}
		});
	}

	@Override
	public void onClick(View v) {
		int id = v.getId();
		if (id == R.id.start_button) {
			onStartButtonClicked();
		}
	}

	private void onStartButtonClicked() {
		if (countrySelected.equals(dataArray[0])) {
			Toast.makeText(getApplicationContext(), "¡Selecciona país!", Toast.LENGTH_SHORT).show();
		} else {
			QuestionsActivity.startActivity(MainActivity.this, countrySelected);
		}
	}
}

File: app/src/main/java/huitca1212/cuantotemide/MainActivity.kt
package huitca1212.cuantotemide

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.ArrayAdapter
import android.widget.Toast
import androidx.activity.viewModels
import androidx.lifecycle.lifecycleScope
import dagger.hilt.android.AndroidEntryPoint
import huitca1212.cuantotemide.databinding.ActivityMainBinding
import kotlinx.coroutines.launch

@AndroidEntryPoint
class MainActivity : BaseActivity() {

    private lateinit var binding: ActivityMainBinding
    private val viewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.bind(findViewById(R.id.mainMainContainer))
        setSupportActionBar(binding.mainAppTopBarLayout.appTopBar)

        binding.mainWelcomeTextView.text = getString(R.string.welcome_text).fromHtml()
        binding.mainStartButton.setOnClickListener { onStartButtonClicked() }

        loadCountriesAndSetupSpinner()
    }

    private fun loadCountriesAndSetupSpinner() {
        lifecycleScope.launch {
            val countries = viewModel.loadCountries()
            val countryNames = countries.map { it.name }
            val adapter = ArrayAdapter<Any?>(this@MainActivity, android.R.layout.simple_spinner_dropdown_item, countryNames)
            binding.mainCountrySpinner.adapter = adapter
            binding.mainCountrySpinner.onItemSelectedListener = object : OnItemSelectedListener {
                override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                    viewModel.selectCountry(position)
                }

                override fun onNothingSelected(arg0: AdapterView<*>?) = Unit
            }
        }
    }

    private fun onStartButtonClicked() {
        val userName = binding.mainNameEditText.text.toString()
        val country = viewModel.selectedCountry
        country?.code?.let {
            QuestionsActivity.startActivity(activity = this, countrySize = country.size, userName = userName)
        } ?: showError()
    }

    private fun showError() {
        Toast.makeText(this, getString(R.string.welcome_chooser_error), Toast.LENGTH_SHORT).show()
    }
}
